sw/device/exts/common/flash_link.ld

```diff
@@ -0,0 +1,115 @@
+/* Copyright lowRISC contributors. */
+/* Licensed under the Apache License, Version 2.0, see LICENSE for details. */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+/**
+ * Linker script for an OpenTitan flash binaries.
+ *
+ * Portions of this file are Ibex-specific.
+ */
+
+OUTPUT_ARCH(riscv)
+GROUP(-lgcc)
+
+/**
+ * Indicate that there are no dynamic libraries, whatsoever.
+ */
+__DYNAMIC = 0;
+
+MEMORY {
+  ram(w)    : ORIGIN = 0x10000000, LENGTH = 0x10000
+  flash(rx) : ORIGIN = 0x20000000, LENGTH = 0x100000
+}
+
+/**
+ * The stack starts at the end of RAM and grows down.
+ */
+_stack_start = ORIGIN(ram) + LENGTH(ram);
+
+SECTIONS {
+  /**
+   * C runtime (CRT) section, containing program initialization code.
+   *
+   * We don't use ENTRY, so the start of |flash| acts as the entrypoint. Since
+   * the CRT is the first thing that needs to run, it goes at the top.
+   */
+  .crt ORIGIN(flash): {
```

Comment:
This probably doesn't matter for now, but eventually we'll probably need the move the beginning of code down a little bit.  We need some kind of header information eventually on top of the images (or somewhere) to help us determine A/B copies. 

@moidx

Reply:
Definitely.  We're also going to need PIE at some point as well to support apps that are loaded beneath BL layers.

Reply:
Yeah, I think this depends on figuring out the header format eventually. The linker can help us take care of that, but for now, we're pretending the header is zero bytes tall and we just jump straight into flash.

Reply:
I decided to actually just deal with this now. I added the barebone-est of headers, consisting of the entry address, mostly because the idea of "oh we just jump to start of flash" rather than jumping to a symbol grosses me out.